src/IconButton/index.jsx

```diff
@@ -14,12 +14,20 @@ const IconButton = React.forwardRef(({
   onClick,
   size,
   variant,
-  iconAs: IconComponent,
+  iconAs,
   isActive,
   ...attrs
 }, ref) => {
   const invert = invertColors ? 'inverse-' : '';
   const activeStyle = isActive ? `${variant}-` : '';
+  if (!iconAs && process.env.NODE_ENV === 'development' && console) {
```

Comment:
We can't really use `withDeprecatedProps` helper here to handle deprecation warning message since there is no way of determining whether default prop value was used or it was explicitly passed to the component (we can figure out some workaround for that but it would make `withDeprecatedProps` ugly I feel like). I think placing the warning here temporarily should be fine.

Also, this is why codecov is failing 😅, but I don't think we should add a test for this.

Reply:
I agree on both accounts 😄